test/Microsoft.AspNetCore.Routing.Tests/EndpointHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    internal static class EndpointHelper
+    {
+        public static MatcherEndpoint CreateMatcherEndpoint(
+            string template,
+            object defaults = null,
+            object requiredValues = null,
+            object constraints = null,
+            int order = 0,
+            string routeName = null,
+            EndpointMetadataCollection metadataCollection = null)
+        {
+            if (metadataCollection == null)
+            {
+                metadataCollection = EndpointMetadataCollection.Empty;
+                if (!string.IsNullOrEmpty(routeName))
+                {
+                    metadataCollection = new EndpointMetadataCollection(new[] { new RouteNameMetadata(routeName) });
+                }
+            }
+
+            // Typically this is how MvcEndpointDataSource does, i.e it creates an endpoint only if all the required
+            // values of an action can satisfy a route template.
+            if (requiredValues != null && defaults == null)
+            {
+                defaults = new RouteValueDictionary(requiredValues);
+            }
```

Comment:
I don't think this should be here. The caller should pass in the data they want.